flowr/src/main/java/com/fueled/flowr/Flowr.java

```diff
@@ -134,6 +147,27 @@ public static ResultResponse getResultsResponse(Bundle arguments, int resultCode
         return resultResponse;
     }
 
+    private Constructor<? extends FlowrDeepLinkHandler> findBindingConstructorForClass() throws ClassNotFoundException, NoSuchMethodException, IllegalAccessException, InvocationTargetException, InstantiationException {
+        //Use reflexion because:
+        //1. The FlowrDeepLinkHandlerImpl location is dynamic based on the location of all the Fragment that support deep link, so we can't rely on a static position like Dagger.
+        //2. Because the FlowrDeepLinkHandlerImpl is generated only when a @DeepLink is used, if the project doesn't use deep linking, the class will never be generated and the project won't compile
+        FlowrConfig config = (FlowrConfig) getClass().getClassLoader().loadClass(FLOWR_CONFIG).getConstructor().newInstance();
+        Class<?> bindingClass = getClass().getClassLoader().loadClass(config.getHandlerPackage());
+        //noinspection unchecked
+        return (Constructor<? extends FlowrDeepLinkHandler>) bindingClass.getConstructor();
+    }
+
+    /**
+     * Set an intent to be parse for Deep Link support.
+     *
+     * @param intent The activity that contains the FlowR instance.
+     * @return this instance for FlowR.
+     */
+    public Flowr setDeepLinkContext(@NonNull Intent intent) {
```

Comment:
@julien-fueled I think we can remove this and add an open method that accepts both an `Intent` and a `Fragment` like you proposed earlier